packages/grid/_modules_/grid/hooks/features/virtualization/useGridVirtualRows.ts

```diff
@@ -193,51 +209,38 @@ export const useGridVirtualRows = (apiRef: GridApiRef): void => {
       logger.debug(`Scrolling to cell at row ${params.rowIndex}, col: ${params.colIndex} `);
 
       const scrollCoordinates: any = {};
-      const isColVisible = apiRef.current.isColumnVisibleInWindow(params.colIndex);
-      logger.debug(`Column ${params.colIndex} is ${isColVisible ? 'already' : 'not'} visible.`);
-      if (!isColVisible) {
-        const isLastCol = params.colIndex + 1 === columnsMeta.positions.length;
-
-        if (isLastCol) {
-          const lastColWidth = visibleColumns[params.colIndex].width!;
-          scrollCoordinates.left =
-            columnsMeta.positions[params.colIndex] +
-            lastColWidth -
-            gridState.containerSizes!.windowSizes.width;
-        } else {
-          scrollCoordinates.left =
-            columnsMeta.positions[params.colIndex + 1] -
-            gridState.containerSizes!.windowSizes.width +
-            gridState.scrollBar!.scrollBarSize.y;
-          logger.debug(`Scrolling to the right, scrollLeft: ${scrollCoordinates.left}`);
-        }
-        if (gridState.rendering.renderingZoneScroll.left > scrollCoordinates.left) {
-          scrollCoordinates.left = columnsMeta.positions[params.colIndex];
-          logger.debug(`Scrolling to the left, scrollLeft: ${scrollCoordinates.left}`);
-        }
+
+      if (params.colIndex != null) {
+        scrollCoordinates.left = scrollIntoView({
+          clientHeight: windowRef.current!.clientWidth,
```

Comment:
I noticed a difference of 1px between `clientWidth` and the [old values](https://github.com/mui-org/material-ui-x/pull/2162/files#diff-7904e8c986f79f8bbc1627613039ba9fc46cc3f000aef6b4883c56f4aff2311aL210-L211). Should we calculate the inner width by ourselves?

```suggestion
          clientHeight: gridState.containerSizes!.windowSizes.width - gridState.scrollBar.scrollBarSize.y,
```

In Chrome, it misses the right border. In Firefox it works fine. This doesn't happen in https://material-ui.com/components/data-grid/demo/

<img width="450" src="https://user-images.githubusercontent.com/42154031/126253044-f936f64c-d210-4bec-bc3c-a5e90e680cb1.gif" alt="DVOMOyM6Lk" style="max-width:100%;">

Reply:
I have tried to reproduce on BS with Windows 10 and Chrome 91 but can't:

<img width="1005" alt="Capture d’écran 2021-07-20 à 11 45 17" src="https://user-images.githubusercontent.com/3165635/126301487-327e9a5b-798a-451d-a3df-44adfacfbfa1.png">

Reply:
> Should we calculate the inner width by ourselves?

I think that it would be better to reduce the number of dependencies here. The more we can rely on direct access to the DOM API, the better. It would be great to get to the root cause of what you have found. Is it a sub-pixel alignment issue? Is it a CSS outline rendering glitch? Did you zoom?

Reply:
> I have tried to reproduce on BS with Windows 10 and Chrome 91 but can't:

I noticed it only happens if the Windows's scaling is above 100%. Here's 125% because the resolution is 1920x1080. To simulate it, open Chrome in BrowserStack and zoom to 125%. Use https://deploy-preview-2162--material-ui-x.netlify.app/components/data-grid/demo/ to test. The `clientWidth` should be 830 but it's 829.

Reply:
I can reproduce with a zoom of 125% on Windows and macOS. I can also reproduce on AG Grid.

<img width="659" alt="Capture d’écran 2021-07-20 à 15 14 08" src="https://user-images.githubusercontent.com/3165635/126330400-b7234c5a-2d95-46d8-b4d8-5df721b9d563.png">

I guess Chrome is rounding numbers when we read the DOM but differently based on what we read.

Reply:
Yes, let's see if some user complains about. With the current logic we use today, this glitch doesn't occur. That was my only concern.

Reply:
In any case, 💯 for having found this. Even if I don't that we should try hard to fix it at the state of the component (high opportunity cost). 

I think that later on, once we need to find new ways to further differentiate in this mature market, it could matter.